test/burn/org/apache/cassandra/net/ConnectionBurnTest.java

```diff
@@ -467,26 +491,42 @@ public void run() throws ExecutionException, InterruptedException, NoSuchFieldEx
                     }
                 });
 
-                while (deadline > nanoTime() && failed.getCount() > 0)
+                int added = 0;
+                while (deadline > nanoTime() && added == 0)
                 {
                     reporters.update();
                     reporters.print();
-                    Uninterruptibles.awaitUninterruptibly(failed, 30L, TimeUnit.SECONDS);
+                    added = Queues.drainUninterruptibly(unexpectedExceptions, exceptions, 10, 30L, TimeUnit.SECONDS);
                 }
 
                 executor.shutdownNow();
                 ExecutorUtils.awaitTermination(5L, TimeUnit.MINUTES, executor);
+
+                if (!exceptions.isEmpty())
+                    throw exceptions.get(0);
```

Comment:
should probably addSuppressed any items > index 0

Reply:
Fixed.